mod-source-record-manager-server/src/main/java/org/folio/dao/JobExecutionDaoImpl.java

```diff
@@ -42,166 +58,216 @@
         throw new NotFoundException(rollbackMessage);
       }
-      Future<Void> endTxFuture = Future.future(); //NOSONAR
-      pgClientFactory.createInstance(tenantId).endTx(tx, endTxFuture);
-      return endTxFuture;
+      Promise<Void> endTxFuture = Promise.promise();
+      pgClientFactory.createInstance(tenantId).endTx(tx.future(), endTxFuture);
+      return endTxFuture.future();
     }).setHandler(v -> {
       if (v.failed()) {
-        pgClientFactory.createInstance(tenantId).rollbackTx(tx, rollback -> future.fail(v.cause()));
+        pgClientFactory.createInstance(tenantId).rollbackTx(tx.future(), rollback -> promise.fail(v.cause()));
         return;
       }
-      future.complete(jobExecutionFuture.result());
+      promise.complete(jobExecutionFuture.result());
     });
-    return future;
+    return promise.future();
   }
 
   @Override
   public Future<Boolean> deleteJobExecutionById(String jobExecutionId, String tenantId) {
-    Future<UpdateResult> future = Future.future();
-    pgClientFactory.createInstance(tenantId).delete(TABLE_NAME, jobExecutionId, future.completer());
-    return future.map(updateResult -> updateResult.getUpdated() == 1);
+    Promise<UpdateResult> promise = Promise.promise();
+    pgClientFactory.createInstance(tenantId).delete(TABLE_NAME, jobExecutionId, promise);
+    return promise.future().map(updateResult -> updateResult.getUpdated() == 1);
+  }
+
+  private SqlSelect getSqlSelect(String tableName, String cql) throws FieldException {
+    try {
+      return new CQL2PgJSON(tableName + ".jsonb").toSql(cql);
+    } catch (QueryValidationException e) {
+      throw new CQLQueryValidationException(e);
+    }
+  }
+
+  private String prepareQueryGetJobWithoutParentMultiple(SqlSelect sqlSelect, int limit, int offset, String schemaName) {
+    String whereClause = String.format("WHERE %s ", sqlSelect.getWhere());
+    String orderBy = sqlSelect.getOrderBy().isEmpty() ? StringUtils.EMPTY : String.format("ORDER BY %s", sqlSelect.getOrderBy());
+    return String.format(getJobsWithoutParentMultipleSql, schemaName, whereClause, schemaName, schemaName, whereClause, orderBy, limit, offset);
+  }
+
+  private JobExecutionCollection mapResultSetToJobExecutionCollection(ResultSet resultSet) {
+    int totalRecords = resultSet.getNumRows() != 0 ? resultSet.getRows(false).get(0).getInteger(TOTAL_ROWS_COLUMN) : 0;
+    List<JobExecution> jobExecutions = resultSet.getRows().stream()
+      .map(row -> mapJsonToJobExecution(row.getString(JSONB_COLUMN)))
+      .collect(Collectors.toList());
+
+    return new JobExecutionCollection()
+      .withJobExecutions(jobExecutions)
+      .withTotalRecords(totalRecords);
+  }
+
+  private JobExecution mapJsonToJobExecution(String jsonString) {
+    try {
+      return ObjectMapperTool.getMapper().readValue(jsonString, JobExecution.class);
+    } catch (IOException e) {
+      LOGGER.error("Error while mapping json to jobExecution", e);
```

Comment:
I don`t think that it is the best place for using anti-pattern "log-and-throw". @OleksiiKuzminov What do you think? Do we really need this logic here?